omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java

```diff
@@ -232,20 +233,33 @@ protected PageableResult doSearch(RequestContext context) {
 			
 			String careSettingUuid = context.getRequest().getParameter("careSetting");
 			String asOfDateString = context.getRequest().getParameter("asOfDate");
+			String orderTypeUuid = context.getRequest().getParameter("orderType");
 			String sortParam = context.getRequest().getParameter("sort");
 			
 			CareSetting careSetting = null;
 			Date asOfDate = null;
+			OrderType orderType = null;
 			if (StringUtils.isNotBlank(asOfDateString)) {
 				asOfDate = (Date) ConversionUtil.convert(asOfDateString, Date.class);
 			}
 			if (StringUtils.isNotBlank(careSettingUuid)) {
 				careSetting = ((CareSettingResource1_10) Context.getService(RestService.class).getResourceBySupportedClass(
 				    CareSetting.class)).getByUniqueId(careSettingUuid);
+				if (careSetting == null) {
```

Comment:
Why are you adding this?

Reply:
The intention is that if no care setting uuid is passed in, return results matching all care settings, but if an invalid care setting is passed in, do not return any results.

The current functionality returns orders in all care settings if an invalid care setting is passed in, which seems incorrect.

Reply:
If an invalid value is passed in, it's misleading to a client to send back an empty list, you should communicate back to them the problem by failing out loud.